lib/puppet/type/openldap_database.rb

```diff
@@ -83,6 +83,26 @@ def insync?(is)
           "{SSHA}" + Base64.encode64("#{Digest::SHA1.digest("#{should}#{salt}")}#{salt}").chomp == is
         when /^\{SHA\}.+/
           "{SHA}" + Digest::SHA1.hexdigest(should) == is
+        when /^\{SSHA(256|384|512)\}.+/
```

Comment:
I did a mistake I should have checked my code more carefully.
I dropped a regex during the copy and paste.